Gems/Atom/Feature/Common/Code/Source/Silhouette/SilhouetteFeatureProcessor.cpp

```diff
@@ -176,30 +184,37 @@ namespace AZ::Render
 
     void SilhouetteFeatureProcessor::OnRenderPipelineChanged(AZ::RPI::RenderPipeline* pipeline, [[maybe_unused]] AZ::RPI::SceneNotification::RenderPipelineChangeType changeType)
     {
-        // update our render pass members in case the pipeline was removed
-        UpdatePasses(pipeline);
+        // Only need to recache silhouette passes if the pipeline had them
+        if (pipeline == m_renderPipeline)
+        {
+            if (changeType == AZ::RPI::SceneNotification::RenderPipelineChangeType::Removed)
```

Comment:
minor - since you are using `changeType` you can remove the `[[maybe_unused]]` on line 185